client/client.go

```diff
@@ -113,7 +114,17 @@ func New(ctx context.Context, address string, opts ...ClientOpt) (*Client, error
 		if err != nil {
 			return nil, err
 		}
-		gopts = append(gopts, grpc.WithContextDialer(dialFn))
+		if dialFn != nil {
+			gopts = append(gopts, grpc.WithContextDialer(dialFn))
+		}
+		addressParts := strings.SplitN(address, "://", 2)
+		if len(addressParts) != 2 {
+			return nil, fmt.Errorf("%s is missing scheme", address)
+		}
+		if addressParts[0] == "tcp" {
+			// remove tcp scheme from address, since default dialer doesn't expect that
+			address = addressParts[1]
```

Comment:
Is this logic still needed? It looks to me like GRPC should probably parse this case: https://github.com/moby/buildkit/blob/080bd42f07cc192ec635ef0dde47654fb62a4e75/vendor/google.golang.org/grpc/internal/transport/http2_client.go#L150-L178

If it is still needed, then note below where we do `url.Parse` to determine the authority, so `address` must remain a valid URL.

Reply:
As I read it here, the parsing of the scheme is only done for "unix://". If no scheme is found, the default (tcp) will be used.

https://github.com/moby/buildkit/blob/cab7db27074d275af1eeeaf12c3bd7bc8266cb88/vendor/google.golang.org/grpc/internal/transport/http_util.go#L387-L411

Also from the grpc-go docs as defined [here](https://github.com/grpc/grpc/blob/master/doc/naming.md) the target address naming doesn't expect the tcp:// scheme to be prefixed.

When removing the logic above, the following error accurs: `Unavailable: connection error: desc = "transport: error while dialing: dial tcp: address tcp://localhost:443: too many colons`

Reply:
I've moved the logic beneath the authority determination